dockeradapter/dockeradapter.go

```diff
@@ -318,3 +403,41 @@ func stripDockerLogsHeader(rawlogs string) string {
 	}
 	return rawlogs[headerLength:]
 }
+
+func (c *concreteDockerClient) CleanupHarvesters(ctx context.Context, key string) error {
+	logging.Debug("[CleanupHarvesters] Forcefully removing attached harvester containers with the key %s", key)
+	filters := filters.NewArgs()
+	filters.Add("label", key)
+
+	listOptions := dockertypes.ContainerListOptions{
+		All:     true,
+		Filters: filters,
+	}
+
+	containers, err := c.Client.ContainerList(ctx, listOptions)
+	if err != nil {
+		return err
```

Comment:
Worth providing some additional context for this error given it's bubbling up from the underlying Docker client?

Reply:
I see the context if provided by the caller. Ignore this.